src/name_bazaar/ui/pages/register_name_page.cljs

```diff
@@ -1,45 +1,175 @@
+
+   })
+
+(defn load-availiability [value]
+  (let [full-name (str value constants/registrar-root)
+        label-hash (sha3 value)]
+    (when (top-level-name? full-name)
+      (dispatch [:registrar.entries/load [label-hash]]))))
 
 (defn register-name-form []
-  (let [label (r/atom "")]
-    (fn []
-      [ui/Grid
-       {:class "layout-grid submit-footer"}
-       [ui/GridRow
-        [ui/GridColumn
-         {:mobile 16
-          :tablet 8
-          :computer 6}
-         [ens-name-input
-          {:label "Name"
-           :fluid true
-           :value @label
-           :on-change (fn [_ data]
-                        (let [value (aget data "value")]
-                          (when (valid-ens-name? value)
-                            (reset! label value))))}]]]
-       [ui/GridRow
-        {:centered true}
-        [:div
-         [transaction-button
-          {:primary true
-           :disabled (empty? @label)
-           :pending-text "Registering..."
-           :on-click (fn []
-                       (when (and (not (empty? @label))
-                                  (valid-ens-name? @label))
-                         (dispatch [:registrar/register {:ens.record/label @label}])
-                         (reset! label "")))}
-          "Register"]]]])))
+  (let [label (r/atom "test111")
+        bid-value (r/atom 0.01)]
+    (fn [{:keys [:editing?]}]
+      (let [disabled? #(or (empty? @label)
+                           (not (valid-ens-name? @label)))
+            label-hash (sha3 @label)
+            state @(subscribe [:registrar/auction-state label-hash])
+            {:keys [:registrar.entry/registration-date]} @(subscribe [:registrar/entry label-hash])]
+
+        [ui/Grid
+         {:class "layout-grid submit-footer"
+          :celled "internally"}
+         [ui/GridRow
+          [ui/GridColumn
+           {:mobile 16
+            :tablet 8
+            :computer 6}
+           [:div.input-state-label
+            {:class (case state
+                      :registrar.entry.state/open :success
+                      :error)}
+            [ens-name-input
+             {:label "Name"
+              :fluid true
+              :value @label
+              :on-change (fn [_ data]
+                           (let [value (aget data "value")]
+                             (when (valid-ens-name? value)
+                               (do
+                                 (reset! label value)
+                                 (load-availiability value)))))}]
+            [:div.ui.label (get registrar-entry-state->text state)]]]
+          [ui/GridColumn
+           {:mobile 16
+            :tablet 8
+            :computer 6
+            :floated "right"
+            :text-align :center}
+           [:div "IMPORT/EXPORT"]]]
+
+         (if (= state :registrar.entry.state/user-made-bid)
+           (let [end-bidding-date (t/minus registration-date (t/hours 48))
```

Comment:
Please move such calculations into a subscription. Components should contain as least logic as possible, and subscriptions are being cached